# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/test/property_database.js
# Commit: 473ebe5ec7aa
# Full Hash: 473ebe5ec7aa4f1c5da45ed580fb56258737a210
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2023-08-23 04:21:33
# Regressor Bug: 1819464
# File Overlap Count: 2
# Description:
#   Bug 1819464 - Part 2: Add content-box and stroke-box into transform-box in style. r=emilio
#   
#   Also, add a prefernce for content-box and stroke-box. So we have to
#   define a new type, TransformBox, and let content-box and stroke-box
#   behind the preference.
# ==============================================================================

diff -r 9c22e16e0567 -r 473ebe5ec7aa layout/style/test/property_database.js
--- a/layout/style/test/property_database.js	Tue Aug 22 19:49:03 2023 +0000
+++ b/layout/style/test/property_database.js	Tue Aug 22 19:49:03 2023 +0000
@@ -3775,7 +3775,7 @@
     type: CSS_TYPE_LONGHAND,
     initial_values: ["view-box"],
     other_values: ["fill-box", "border-box"],
-    invalid_values: ["content-box", "padding-box", "stroke-box", "margin-box"],
+    invalid_values: ["padding-box", "margin-box"],
   },
   "transform-origin": {
     domProp: "transformOrigin",
@@ -11996,6 +11996,15 @@
   };
 }
 
+if (
+  IsCSSPropertyPrefEnabled("layout.css.transform-box-content-stroke.enabled")
+) {
+  gCSSProperties["transform-box"]["other_values"].push(
+    "content-box",
+    "stroke-box"
+  );
+}
+
 gCSSProperties["touch-action"] = {
   domProp: "touchAction",
   inherited: false,